### PR TITLE
Adding the A. Dux keyboard

### PR DESCRIPTION
**Pull request by Albert-IV**
This adds support for the A. Dux keyboard by @tapioki to QMK.  Let me know if there are any questions or concerns!

## Description

Adds everything someone would need to flash their A. Dux with working QMK firmware.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/a_dux/a_dux.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+// readability
+#define ___ KC_NO
+
+#define LAYOUT( \
+        L01, L02, L03, L04, L05, R01, R02, R03, R04, R05, \
+        L06, L07, L08, L09, L10, R06, R07, R08, R09, R10, \
+        L11, L12, L13, L14, L15, R11, R12, R13, R14, R15, \
+                  L16, L17, R16, R17       \
```
Comment (drashna) [inline keyboards/a_dux/a_dux.h:35]: visual nitpick: 
```suggestion
        L11, L12, L13, L14, L15, R11, R12, R13, R14, R15, \
                       L16, L17, R16, R17       \
```

File: keyboards/a_dux/keymaps/default/readme.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+A usable default keymap for the Ferris keyboard
```
Comment (drashna) [inline keyboards/a_dux/keymaps/default/readme.md:1]: Isn't the ferris?

File: keyboards/a_dux/keymaps/jcmkk3/keymap.c [hunk 3]
```diff
@@ -0,0 +1,115 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/a_dux/keymaps/jcmkk3/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/a_dux/keymaps/jcmkk3/keymap_bak.c [hunk 4]
```diff
@@ -0,0 +1,109 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/a_dux/keymaps/jcmkk3/keymap_bak.c:1]: These extra files should be removed.

File: keyboards/a_dux/keymaps/jcmkk3/oneshot.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#pragma once
```
Comment (drashna) [inline keyboards/a_dux/keymaps/jcmkk3/oneshot.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/a_dux/keymaps/jcmkk3/oneshot.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
+#include "oneshot.h"
```
Comment (drashna) [inline keyboards/a_dux/keymaps/jcmkk3/oneshot.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/a_dux/keymaps/jcmkk3/rules.mk [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Bootloader selection
+BOOTLOADER = dfu
```
Comment (drashna) [inline keyboards/a_dux/keymaps/jcmkk3/rules.mk:2]: This shouldn't be needed, and sets it to an invalid bootloader type.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/a_dux/keymaps/default/readme.md (hunk 2 above)
Comment (droppedoncaprica) [reply keyboards/a_dux/keymaps/default/readme.md:1]: Ope, missed that! A lot of this config was sourced from the Ferris, including the default keymap. I'll get this fixed!

File: keyboards/a_dux/keymaps/jcmkk3/keymap.c (hunk 3 above)
Comment (droppedoncaprica) [reply keyboards/a_dux/keymaps/jcmkk3/keymap.c:1]: Sure can!

File: keyboards/a_dux/a_dux.h (hunk 1 above)
Comment (droppedoncaprica) [reply keyboards/a_dux/a_dux.h]: Fixed!

**Comment by droppedoncaprica** [comment]
Addressed the issues brought up in the PR.  I left the default keymap that leverages the Ferris' default keymap for now.  

That being said, I can refactor it to be a super basic base layout for others to extend if you'd prefer to see it like that.

File: keyboards/a_dux/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Architeuthis Dux
```
Comment (drashna) [inline keyboards/a_dux/README.md:1]: Please rename this file to be lower case.

File: keyboards/a_dux/README.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+* [Tented Case](https://github.com/jdart/adux-tent)
+* [Case with Plates](https://github.com/sadekbaroudi/cephalopoda/tree/main/Architeuthis%20dux/case)
+
+### Keyboard Info
+
+* Keyboard Maintainer: [tapioki](https://github.com/tapioki)
+* Hardware Supported: Architeuthis Dux
+* Hardware Availability: [GitHub](https://github.com/tapioki/cephalopoda/tree/main/Architeuthis%20dux)
+
+Make examples for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/a_dux/README.md:23]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/a_dux/README.md (hunk 8 above)
Comment (drashna) [inline keyboards/a_dux/README.md:1]: Could you rename this file to be lower case?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/a_dux/README.md (hunk 9 above)
Comment (droppedoncaprica) [reply keyboards/a_dux/README.md]: This keyboard doesn't have a physical reset button, so I described how to reset the microcontroller by shorting the reset and round pins.  Let me know if that works!

**Review by drashna — APPROVED** [review]

File: keyboards/a_dux/rules.mk [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes        # Use shared split_common code
```
Comment (fauxpark) [inline keyboards/a_dux/rules.mk:27]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
UNICODE_ENABLE = yes
SPLIT_KEYBOARD = yes
```

File: keyboards/a_dux/keymaps/jcmkk3/rules.mk [hunk 11]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (fauxpark) [inline keyboards/a_dux/keymaps/jcmkk3/rules.mk:1]:
```suggestion
```

File: keyboards/a_dux/info.json [hunk 12]
```diff
@@ -0,0 +1,54 @@
+{
+    "keyboard_name": "Architeuthis dux",
+    "url": "https://github.com/tapioki/cephalopoda",
+    "maintainer": "@tapioki",
+    "width": 12,
+    "height": 4.75,
```
Comment (fauxpark) [inline keyboards/a_dux/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by droppedoncaprica** [comment]
Anything else we need? Let me know if you have any other changes you all would like to see!

File: keyboards/a_dux/rules.mk [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = yes
+SPLIT_KEYBOARD = yes
```
Comment (fauxpark) [inline keyboards/a_dux/rules.mk:25]:
```suggestion
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
UNICODE_ENABLE = yes
SPLIT_KEYBOARD = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Albert-IV** [comment]
@fauxpark

File: keyboards/a_dux/readme.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    make a_dux:default:avrdude-split-right
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+
+* **Physical reset**: Briefly short the RST and GND pins on the microcontroller using tweezers, a paperclip, or any other conductive material.
+
```
Comment (drashna) [inline keyboards/a_dux/readme.md:37]:
```suggestion
* **Physical reset**: Briefly short the RST and GND pins on the microcontroller using tweezers, a paperclip, or any other conductive material.
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by daliusd** [comment]
Tested and this works with my keyboard.

File: keyboards/a_dux/readme.md (hunk 14 above)
Comment (Albert-IV) [reply keyboards/a_dux/readme.md]: These changes have been pulled in.  Sorry for the delay!

**Comment by drashna** [comment]
Thanks!

**Pull request merged**